app/pages/project/index.cjsx

```diff
@@ -87,8 +88,12 @@ ProjectPage = React.createClass
       @context.geordi?.remember projectToken: nextProps.project?.slug
 
     # Only call to get workflow if we know if there is a user or not and the project is finished loading
-    if nextContext.initialLoadComplete and not nextProps.loading and nextProps.preferences and @state.activeWorkflows.length is 0
-      @getAllWorkflows(nextProps.project)
+    if nextContext.initialLoadComplete and not nextProps.loading and nextProps.preferences and @state.activeWorkflows.length is 0 and not @state.loadingSelectedWorkflow
```

Comment:
Ditto what Cam said. The logic here is hard to understand. Is it possible to at least separate out the loading code somehow so that we don't have to worry here about whether or not the component is loading data?